### ISSUE TITLE
Time64Col incorrectly stores nan values as zeros

### ISSUE THREAD

**Issue by cberzan** [PyTables/PyTables#410]
If I write a `nan` value in a `Time64Col`, I get back a `0.0` instead of a `nan`. Minimal example:

``` python
import tables


class Item(tables.IsDescription):
    dt = tables.Time64Col(dflt=float('nan'))


if __name__ == "__main__":
    path = "file.h5"

    # Write table:
    h5file = tables.open_file(path, mode='w')
    group = h5file.create_group('/', 'group1')
    table = h5file.create_table(group, 'table1', Item)
    row = table.row
    row.append()
    row['dt'] = float('nan')
    row.append()
    table.flush()
    h5file.close()

    # Read table back:
    h5file = tables.open_file(path, mode='r')
    table = h5file.root.group1.table1
    for row in table.iterrows():
        print row['dt']
    h5file.close()
```

This prints two `0.0`s instead of two `nan`s. If I change Time64Col to Float64Col, I get two `nan`s, as expected. This strange behavior of Time64Col should either be fixed, or documented. (The documentation suggests that Time64Col behaves exactly like Float64Col.)

My `tables.__version__` is 3.1.1 and `tables.hdf5_version` is 1.8.4-patch1.

**Comment by FrancescAlted** [comment]
Yes, there is an explicit conversion between HDF5 time64 represenation and NumPy that is made in:

https://github.com/PyTables/PyTables/blob/develop/tables/hdf5extension.pyx#L1139

This needs to be fixed.  Willing to provide a pull request?

**Comment by andreabedini** [comment]
H5T_TIME format is not documented anywhere apparently. HDF5 docs states

> The time datatype, H5T_TIME, has not been fully implemented and is not supported. If H5T_TIME is used, the resulting data will be readable and modifiable only on the originating computing platform; it will not be portable to other platforms.

I don't know how to fix this.
